relayer-cli/src/commands/tx/channel.rs

```diff
@@ -0,0 +1,389 @@
+            .chains
+            .iter()
+            .find(|c| c.id == self.dest_chain_id.parse().unwrap())
+            .ok_or_else(|| "missing destination chain configuration".to_string())?;
+
+        let src_chain_config = config
+            .chains
+            .iter()
+            .find(|c| c.id == self.src_chain_id.parse().unwrap())
+            .ok_or_else(|| "missing src chain configuration".to_string())?;
+
+        let signer_seed = std::fs::read_to_string(&self.seed_file).map_err(|e| {
+            anomaly::Context::new("invalid signer seed file", Some(e.into())).to_string()
+        })?;
+
+        let opts = ChannelOpenOptions {
+            dest_chain_config: dest_chain_config.clone(),
+            src_chain_config: src_chain_config.clone(),
+
+            dest_connection_id: self.dest_connection_id.clone(),
+
+            dest_port_id: self.dest_port_id.clone(),
+            src_port_id: self.src_port_id.clone(),
+
+            dest_channel_id: self.dest_channel_id.clone(),
+            src_channel_id: self.src_channel_id.clone(),
+
+            ordering: self.ordering,
+            signer_seed,
+        };
+
+        Ok(opts)
+    }
+}
+
+impl Runnable for TxRawChanAckCmd {
+    fn run(&self) {
+        let config = app_config();
+
+        let opts = match self.validate_options(&config) {
+            Err(err) => {
+                status_err!("invalid options: {}", err);
+                return;
+            }
+            Ok(result) => result,
+        };
+        status_info!("Message", "{:?}", opts);
+
+        let res: Result<String, Error> =
+            build_chan_ack_and_send(&opts).map_err(|e| Kind::Tx.context(e).into());
+
+        match res {
+            Ok(receipt) => status_info!("channel ack, result: ", "{:?}", receipt),
+            Err(e) => status_info!("channel ack failed, error: ", "{}", e),
+        }
+    }
+}
+
+#[derive(Clone, Command, Debug, Options)]
+pub struct TxRawChanConfirmCmd {
+    #[options(free, help = "identifier of the destination chain")]
+    dest_chain_id: String,
+
+    #[options(free, help = "identifier of the source chain")]
+    src_chain_id: String,
+
+    #[options(free, help = "identifier of the destination connection")]
+    dest_connection_id: ConnectionId,
+
+    #[options(free, help = "identifier of the destination port")]
+    dest_port_id: PortId,
+
+    #[options(free, help = "identifier of the source port")]
+    src_port_id: PortId,
+
+    #[options(free, help = "identifier of the destination channel")]
+    dest_channel_id: ChannelId,
+
+    #[options(free, help = "identifier of the source channel")]
+    src_channel_id: ChannelId,
+
+    #[options(help = "the channel order", short = "o")]
+    ordering: Order,
+
+    #[options(
+        help = "json key file for the signer, must include mnemonic",
+        short = "k"
+    )]
+    seed_file: String,
+}
+
+impl TxRawChanConfirmCmd {
+    fn validate_options(&self, config: &Config) -> Result<ChannelOpenOptions, String> {
```

Comment:
The three `validate_options` methods for try, ack, and confirm are identical. That's some potential to improve when we refactor this (unless you prefer to do it now). There is overlap across the `TxRaw..Cmd` structs also.

Reply:
Yes, it's the same as for connection CLIs. I am planning to look at all in a different PR.